evm/src/main/java/com/horizen/evm/LibEvm.java

```diff
@@ -0,0 +1,154 @@
+package com.horizen.evm;
+
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.horizen.evm.interop.*;
+import com.horizen.evm.utils.Address;
+import com.horizen.evm.utils.Hash;
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+
+import java.math.BigInteger;
+
+final class LibEvm {
```

Comment:
For gas estimation, is the intention to "throw the state chages away" on the java side and simply use evmExecute?

Reply:
For gas estimation I expect to re-use parts or even all of the geth implementation. So I think it makes more sense to do that on the go-library side.